pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -473,6 +467,12 @@ func (r *ReconcilePerconaServerMongoDB) Reconcile(request reconcile.Request) (re
 		return rr, err
 	}
 
+	err = r.checkFinalizers(cr)
```

Comment:
we can't move this check there, because when cluster is being deleted we need to check finalizers and do required work.
In this case, we could never reach this code part

Reply:
Finalizer list needs to be empty for resource to be deleted, doesn't it? As far as I see, we only set finalizers in `r.checkFinalizers`

Reply:
Yes. But for example form PXC.
We have `delete pxc pods in order`
Which means, that pxc will be unavailable because of the pods deletion.
And we will never reach this code part, because we will fail somewhere at the beginning while connecting to database.
Operator should not perform any logic when instance is being deleted